pkgs/by-name/gx/gxml/package.nix

```diff
@@ -0,0 +1,58 @@
+{ fetchFromGitLab
+, glib
+, gobject-introspection
+, meson
+, ninja
+, pkg-config
+, stdenv
+, lib
+, libgee
+, libxml2
+, vala
+}:
+stdenv.mkDerivation rec {
+  name = "gxml";
+  version = "0.20.3";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "gxml";
+    rev = version;
+    hash = "sha256-GlctGxsLyQ2kPV3oBmusRiouG4PPncBTh3vgxhVaQOo=";
+  };
+
+  outputs = [ "out" "dev" "devdoc" ];
+
+  PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_GIRDIR = "${placeholder "dev"}/share/gir-1.0";
+  PKG_CONFIG_GOBJECT_INTROSPECTION_1_0_TYPELIBDIR = "${placeholder "out"}/lib/girepository-1.0";
+
+  doCheck = true;
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [
+    meson
+    pkg-config
+    ninja
+    vala
+    gobject-introspection
+  ];
+
+  buildInputs = [
+    libxml2
+    glib
+    libgee
+  ];
+
+  propagatedBuildInputs = buildInputs;
```

Comment:
You do not need to duplicate the `buildInputs`, `propagatedBuildInputs` already set `buildInputs`.